bundles/org.openhab.binding.sonos/src/main/resources/OH-INF/thing/channels.xml

```diff
@@ -45,6 +45,13 @@
 		<description>Suppress all songs from the current queue</description>
 	</channel-type>
 
+	<channel-type id="codec" advanced="true">
+		<item-type>String</item-type>
+		<label>Codec</label>
+		<description>Name of codec currently being decoded</description>
+		<state readOnly="true"/>
```

Comment:
List of options to be added here.
Add the tag telling that the state is not limited to the list of options (because your list of known codecs is probably partial).